builder/llpcBuilderReplayer.cpp

```diff
@@ -268,6 +268,82 @@ Value* BuilderReplayer::ProcessCall(
             return m_pBuilder->CreateSMod(args[0], args[1]);
         }
 
+    case BuilderRecorder::Tan:
+        {
+            return m_pBuilder->CreateTan(args[0]);
+        }
+
+    case BuilderRecorder::ASin:
+        {
+            return m_pBuilder->CreateASin(args[0]);
+        }
+
+    case BuilderRecorder::ACos:
+        {
+            return m_pBuilder->CreateACos(args[0]);
+        }
+
+    case BuilderRecorder::ATan:
+        {
+            return m_pBuilder->CreateATan(args[0]);
+        }
+
+    case BuilderRecorder::ATan2:
+        {
+            return m_pBuilder->CreateATan2(args[0],
+                       args[1]);
```

Comment:
You can write this in one line

Reply:
We should move to auto-generating all this boilerplate code.

Reply:
I don't want to introduce autogeneration overly. After we make large builder changes, builder will be stable for a long period of time without too many interface methods added. At that time, code review is able to catch those minor issues.

This is similar to image functions. They were implemented in python file and were primarily auto generated. We can see the logic clearly. After they were implemented by builder hand-written codes, it is more clear.

I think auto-generation is suitable for table-driven headers. Otherwises, it makes codes blur.

Reply:
It is not really similar to image functions. They used to have the actual implementation auto-generated by the python script. The code here is boilerplate code that is nothing to do with the implementation of (e.g.) CreateATan2.

Reply:
We assume the replayer just does trivial things. If the replayer will modify data and insert something special to handle or patch certain opcodes, auto-generation file is difficult to achieve this. This is why I don't like autogen file to construct codes rather than uniform table data.